packages/aws-cdk/lib/commands/doctor.ts

```diff
@@ -68,3 +68,10 @@ function displayCdkEnvironmentVariables() {
   }
   return healthy;
 }
+
+function anonymizeAwsVariable(name: string, value: string) {
+  if (name === 'AWS_ACCESS_KEY_ID') { return value.substr(0, 4) + '*'.repeat(Math.max(0, value.length - 4)); }
+  if (name === 'AWS_SECRET_ACCESS_KEY') { return '*'.repeat(value.length); }
+  if (name === 'AWS_SESSION_TOKEN') { return '*'.repeat(50); }
```

Comment:
No strong feelings, but I'd usually literally output `<redacted>` for those instead of a string of `*`s.